### PR TITLE
Mock tests (initial approach)

### LINKED ISSUE #10: Improve testing on formatting of Metrics object
Currently we have few tests around the formatting of the metrics output from Prometheus, this should be improved.

We should also avoid having a reliance on external systems for the tests. Perhaps we can create a data set for use in testing?

[test-metrics-format.R](https://github.com/glenn-m/promR/blob/master/tests/testthat/test-metrics-format.R)

### PR DESCRIPTION
**Pull request by konradzdeb**
Hi,

This doesn't fully close #10 but I wanted to show you the approach before replacing the rest of tests. The idea is a follows:

1. As per [previoulsy linked chat](https://stackoverflow.com/a/41777777/1655567) I've committed RDS object containing instant response from `httr::GET`
2. `Stub` for `httr::GET` is created via `mockery` package

This initial test on a returned object being data frame passes. I reckon that this could be further developed.

### CODE REVIEW

File: vignettes/basic-usage.Rmd [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+Prometheus is an open-source software project recording real-time metrics using HTTP pull model [^1]. Prometheus records metrics in a time series database and allows for high-dimensionality. 
+
+## Queries
+
+The `promR` package offers `Promethous` class that provides two methods of sourcing data:
```
Comment (glenn-m) [inline vignettes/basic-usage.Rmd]: Typo here, `Promethous`

File: tests/testthat/test-response-check.R [hunk 2]
```diff
@@ -72,8 +72,22 @@ test_that(desc = "OK when application type is json and status code is 200.",
...
 # Test whether package can pull some sample data
 # Source query data once to use later across tests
-prom <-
-  Prometheus$new(host = "http://demo.robustperception.io", port = 9090)
+
+# Create an object
+prom <- Prometheus$new(host = "http://demo.robustperception.io",
+                       port = 9090)
+
+test_that(desc = "Check that class functions source data.",
```
Comment (glenn-m) [inline tests/testthat/test-response-check.R:80]: The approach looks good, removes the need to call out to a service which may go down. I reckon we can roll out this pattern to the other main API calls.

**Review by glenn-m — CHANGES_REQUESTED** [review]
It looks good apart from the Vignette which looks unfinished, might be worthwhile submitting it under a different PR once it's done?

**Comment by konradzdeb** [comment]
Sorry, I’ve mistakenly committed this vignette to a wrong branch!

--
Konrad

File: vignettes/basic-usage.Rmd (hunk 1 above)
Comment (konradzdeb) [reply vignettes/basic-usage.Rmd]: :)

**Review by glenn-m — APPROVED** [review]

**Pull request merged**